spec/requests/api/v1/journeys_controller_update_spec.rb

```diff
@@ -0,0 +1,178 @@
+
+    let(:journey_params) {
+      {
+        data: {
+          "type": 'journeys',
+          "attributes": {
+            "billable": billable,
+            "timestamp": timestamp,
+            "vehicle": { "id": '9876', "registration": 'XYZ' },
+          },
+        },
+      }
+    }
+
+    before do
+      patch "/api/v1/moves/#{move_id}/journeys/#{journey_id}", params: journey_params, headers: headers, as: :json
+      journey.reload
+    end
+
+    context 'when successful' do
+      let(:schema) { load_yaml_schema('get_journey_responses.yaml') }
+
+      context 'when updating all attributes' do
+        it_behaves_like 'an endpoint that responds with success 200'
+
+        it 'returns the correct data' do
+          expect(response_json).to include_json(data: {
+            "id": journey.id,
+            "type": 'journeys',
+            "attributes": {
+              "billable": billable,
+               "vehicle": { "id": '9876', "registration": 'XYZ' },
+            },
+          })
+        end
+
+        it 'updates the underlying journey billable' do
+          expect(journey.billable).to be true
+        end
+
+        it 'updates the underlying journey vehicle' do
+          expect(journey.vehicle).to eql('id' => '9876', 'registration' => 'XYZ')
+        end
+
+        it 'does not update the timestamp' do
+          expect(journey.client_timestamp).to eql Time.zone.parse('2020-05-04T08:00:00Z')
+        end
+      end
+
+      context 'when only updating billable' do
+        let(:journey_params) {
+          {
+            data: {
+              "type": 'journeys',
+              "attributes": {
+                "billable": billable,
+                "timestamp": timestamp,
+              },
+            },
+          }
+        }
+
+        it 'updates the underlying journey billable' do
+          expect(journey.billable).to be true
+        end
+
+        it 'does not update the underlying journey vehicle' do
+          expect(journey.vehicle).to eql('id' => '12345678ABC', 'registration' => 'AB12 CDE')
+        end
+      end
+
+      context 'when only updating the vehicle' do
+        let(:journey_params) {
+          {
+            data: {
+              "type": 'journeys',
+              "attributes": {
+                "vehicle": { "id": '9876', "registration": 'XYZ' },
+                "timestamp": timestamp,
+              },
+            },
+          }
+        }
+
+        it 'does not update the underlying journey billable' do
+          expect(journey.billable).to be false
+        end
+
+        it 'does update the underlying journey vehicle' do
+          expect(journey.vehicle).to eql('id' => '9876', 'registration' => 'XYZ')
+        end
+      end
+    end
+
+
+    context 'when unsuccessful' do
+      let(:schema) { load_yaml_schema('error_responses.yaml') }
+
+      context 'with a bad request' do
+        let(:journey_params) { nil }
+
+        it_behaves_like 'an endpoint that responds with error 400'
```

Comment:
it might be a 422

Reply:
this is a general things we should check for the other responses in these endpoints, maybe we should catch up with Jonny about this to confirm.
I'm telling this because I had the same issue in another endpoint I wrote.

Reply:
so I'm going to leave a blank request `''` as 400 for now as I think that is a different sort of bad request to one where all the parameters are specified, but there is a mistake, e.g. `{"id": "123", "timestamp": "xxx"}`

Reply:
it's ok if you want to leave as it is, I don't think this is very important after all.
I am still not sure because 400 represents "The request could not be understood by the server due to malformed syntax" (https://stackoverflow.com/questions/16133923/400-vs-422-response-to-post-of-data)